src/audio/dai-zephyr.c

```diff
@@ -250,13 +252,69 @@ static enum dma_cb_status dai_dma_cb(struct dai_data *dd, struct comp_dev *dev,
 
 	local_buf = buffer_acquire(dd->local_buffer);
 
-	if (dev->direction == SOF_IPC_STREAM_PLAYBACK)
+	if (dev->direction == SOF_IPC_STREAM_PLAYBACK) {
 		ret = dma_buffer_copy_to(local_buf, dma_buf,
 					 dd->process, bytes);
-	else
-		ret = dma_buffer_copy_from(dma_buf, local_buf,
-					   dd->process, bytes);
+		buffer_release(local_buf);
+	} else {
+		struct list_item *sink_list;
+
+		audio_stream_invalidate(&dma_buf->stream, bytes);
+		ret = dma_buffer_copy_from_no_consume(dma_buf, local_buf, dd->process, bytes);
```

Comment:
ignoring errors?

Reply:
@lyakh the pcm converter functions can never fail. I've added a comment now that says that. So its OK to ignore and move on.

Reply:
ok, this might really be minor - but I really wouldn't even assign `ret` if I know I'll ignore it. There are many functions returning non-void that get called with no return value assignment. I find that preferable over assigning a value and never using it...